cpp_test/ImageCap.cpp

```diff
@@ -0,0 +1,57 @@
+#include <iostream>
+#include <opencv2/opencv.hpp>
+
+using namespace cv;
+using namespace std;
+
+class ImageCap
+{
+    private:
+        Mat frame;
+
+    public:
+        ImageCap(int port)
+        {
+            capImage(port);
+        }
+
+        void capImage(int port)
+        {
+            VideoCapture cap(port);
+
+            bool capSuccess = cap.read(frame);
+
+            if (!capSuccess)
+            {
+                cout << "Error capturing image" << endl;
+            }
+
+            writeImage(frame);
+        }
+
+        void writeImage(Mat image)
+        {
+            bool isSuccess = imwrite("./capture.jpg", image);
```

Comment:
It's probably better to have this function take a file name as well, then call it with the default filename elsewhere. Keeping it flexible would be best.